src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -39,12 +45,14 @@ class symbol_factoryt
     std::vector<const symbolt *> &_symbols_created,
     symbol_tablet &_symbol_table,
     const source_locationt &loc,
-    const c_object_factory_parameterst &object_factory_params)
+    const c_object_factory_parameterst &object_factory_params,
+    std::map<irep_idt, irep_idt> &deferred_array_sizes)
```

Comment:
Where is this passed in as a non-empty map?

Reply:
This is kept as state between calls to `c_nondet_symbol_factory` here: https://github.com/diffblue/cbmc/pull/3572/commits/25be3699b8d0697498e4f4b95915ea4e18ada03b#diff-ffd0a317958e24d1f65f21ed40f67c99L39